### ISSUE TITLE
How to make it work with polymorphic relation?

### ISSUE THREAD

**Issue by GDanielRG** [Askedio/laravel-soft-cascade#31]
I have the following models:

https://gist.github.com/GDanielRG/2fa94cce20898439943e6ac7813c07fd

When I run $user->delete(); it gives the following error:

Askedio\SoftCascade\Exceptions\SoftCascadeLogicException with message 'SQLSTATE[42S22]: Column not found: 1054 Unknown column 'ticket_events.' in 'where clause' (SQL: select count(*) as aggregate from `ticket_events` where `ticket_events`.`` in (o9ractuoan, 12m8237uvk, 16p1b650kk, 89508qr3k9, o0tr8hdnvr))'

I realized that the error goes away if I remove the 'ticketEventable' relation from the $softCascade array in the TicketEvent.php

so if I comment out the following line (from TicketEvent.php):

`protected $softCascade = ['ticketEventable'];`

the delete goes through correctly, it deletes the user, then deletes the tickets related to the user, also the ticket events of the user and the ticket events of the tickets of the user, but the problem is that since I comented that line Im not getting the models from the ticketEventable relationship deleted :/

Any help would be much appreciated

**Comment by maguilar92** [comment]
There is no support for this type of relationship.

I'll look to add it as soon as possible.

**Comment by GDanielRG** [comment]
Awesome! thanks!

**Comment by maguilar92** [comment]
Release published, now you can make polymorphic relations :)

**Comment by GDanielRG** [comment]
I am having issues with the polymorphic cascade delete. It was working fine I don't know what happened now I am getting this error:

(1/1) SoftCascadeLogicException
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'ticket_events.' in 'where clause' (SQL: select count(*) as aggregate from `ticket_events` where `ticket_events`.`` in (1vno3cl3eh, 6beino3ifa, 7bljog8rhe, dq15c32bg0, ehfr0geo2e, figlskbiom, km3chaji8r, pn50iqfv8, q9pj13ojqm, t0km806acm))

Here's a link to my code files: https://gist.github.com/GDanielRG/168858539128e35c0e7a4b1462d58929

Basically, I have a Ticket that has TicketEvents and a TicketEvent can be a TicketReference or a TicketStatusChange or a TicketReply. When I call the delete method on a ticket, I get the error above, if I remove the softcascade of the polymorphic relation ar TicketEvent.php the error is gone except the polymorphic models wont get deleted :(

can you help?

**Comment by GDanielRG** [comment]
Help? :(

**Comment by GDanielRG** [comment]
It appears that the problem is my model having two words in the class name. How should the relation be named for it to work?
